pkg/scheduler/factory/factory.go

```diff
@@ -135,6 +135,10 @@ type Config struct {
 
 	// Disable pod preemption or not.
 	DisablePreemption bool
+
+	// DeleteNominatedPodIfExists is called when a pod is assumed
+	// It will delete the pod from internal cache if it's a nominated pod
+	DeleteNominatedPodIfExists func(pod *v1.Pod)
```

Comment:
I would still prefer exposing the scheduling queue here. This function looks too low level to me to get exposed in the scheduler config. We have the scheduler cache, event recorder, volume binder, etc. here. The scheduling queue looks more of the same nature of the existing members of this struct.